evalml/tests/automl_tests/test_automl.py

```diff
@@ -1899,6 +1899,36 @@ def test_search_with_text(mock_fit, mock_score):
     assert automl.rankings['pipeline_name'][1:].str.contains('Text').all()
 
 
+@pytest.mark.parametrize("problem_type,pipeline_name,ensemble_name",
+                         [('binary', 'Stacked Ensemble Classification Pipeline', 'Stacked Ensemble Classifier'),
+                          ('regression', 'Stacked Ensemble Regression Pipeline', 'Stacked Ensemble Regressor')])
+@patch('evalml.pipelines.BinaryClassificationPipeline.score', return_value={"Log Loss Binary": 0.8})
+@patch('evalml.pipelines.RegressionPipeline.score', return_value={"R2": 0.8})
+def test_search_with_text_and_ensembling(mock_score_reg, mock_score_bin, problem_type, pipeline_name, ensemble_name):
+    X = pd.DataFrame(
+        {'col_1': ['I\'m singing in the rain! Just singing in the rain, what a glorious feeling, I\'m happy again!',
+                   'In sleep he sang to me, in dreams he came... That voice which calls to me, and speaks my name.',
+                   'I\'m gonna be the main event, like no king was before! I\'m brushing up on looking down, I\'m working on my ROAR!',
+                   'In sleep he sang to me, in dreams he came... That voice which calls to me, and speaks my name.',
+                   'In sleep he sang to me, in dreams he came... That voice which calls to me, and speaks my name.',
+                   'I\'m singing in the rain! Just singing in the rain, what a glorious feeling, I\'m happy again!',
+                   'do you hear the people sing? Singing the songs of angry men\n\tIt is the music of a people who will NOT be slaves again!',
+                   'I dreamed a dream in days gone by, when hope was high and life worth living',
+                   'Red, the blood of angry men - black, the dark of ages past',
+                   'do you hear the people sing? Singing the songs of angry men\n\tIt is the music of a people who will NOT be slaves again!',
+                   'Red, the blood of angry men - black, the dark of ages past',
+                   'It was red and yellow and green and brown and scarlet and black and ochre and peach and ruby and olive and violet and fawn...']
+         })
+    if problem_type == 'binary':
+        y = [0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 0]
+    else:
+        y = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type=problem_type, allowed_model_families=["catboost", "xgboost"],
+                          max_batches=4, ensembling=True)
+    automl.search()
+    assert automl.full_rankings.loc[automl.full_rankings['pipeline_name'] == pipeline_name].parameters.iloc[0][ensemble_name]['n_jobs'] == 1
```

Comment:
@ParthivNaresh how long does this test take to run? With the `fit` and `score` mocking I bet its not too long.

An alternative organization would be to move the direct coverage down to `IterativeAlgorithm`:
* Move this test to instead cover `IterativeAlgorithm` directly. Set `ensembling` and `text_in_ensembling` to true, run a few batches and assert that the ensemble pipeline at the end has `n_jobs==1` in its parameters. Basically, what this tests already does.
* Add `AutoMLSearch` coverage which mocks the `IterativeAlgorithm` constructor, initializes `AutoMLSearch` and asserts that during initialization `IterativeAlgorithm` was given the right value for `text_in_ensembling`. Parametrize the test to use a grid on dataset with vs without a text feature and on `text_in_ensembling` set to true vs false.

If the current test runs longer than a few seconds, or if you like this organization more, let's do that!

Reply:
Ah, and I just saw you basically already have the first point done, below! Very cool. In that case, I think we should shrink this test to only cover the plumbing of the `text_in_ensembling` parameter, rather than checking the output of `AutoMLSearch.search`.

Reply:
So I actually chose not to mock `fit` here because that's where the error takes place, so I wanted to make sure we caught that. As you mentioned the first point is covered in the `IterativeAlgorithm` tests. I like the idea of mocking the IterativeAlgorithm as per your second point!

Reply:
And this test does take too long for my liking so I'm sure it will speed it up